galaxy_importer/collection.py

```diff
@@ -49,10 +50,17 @@ def import_collection(file, filename=None, logger=None):
 
 
 def _import_collection(file, filename, logger):
-    with tempfile.TemporaryDirectory() as extract_dir:
+    with tarfile.open(fileobj=file, mode='r') as pkg_tar:
+        manifest_file = pkg_tar.extractfile('MANIFEST.json')
+        manifest_data = json.load(manifest_file)
```

Comment:
Check for an exception here in case `MANIFEST.json` is malformed. It's possible a user may muck with the file or a bug gets introduced in the client.